File: RecentMistakesFirstSorter.ts
import { Card } from "./Card";
import { CardOrganizer } from "./CardOrganizer";

export function newRecentMistakesFirstSorter(): CardOrganizer {
  return {
    organize(cards: Card[]): Card[] {

      const incorrect: Card[] = [];
      const correct: Card[] = [];

      for (const card of cards) {
        if (card.wasAnsweredCorrectlyLastRound()) {
          correct.push(card);
        } else {
          incorrect.push(card);
        }
      }

      return [...incorrect, ...correct];
    }
  };
}

File: RecentMistakesFirstSorter.java
import java.util.ArrayList;
import java.util.List;

public class RecentMistakesFirstSorter implements CardOrganizer {

    @Override
    public List<Card> organize(List<Card> cards) {

        List<Card> incorrect = new ArrayList<>();
        List<Card> correct = new ArrayList<>();

        for (Card card : cards) {
            if (card.wasAnsweredCorrectlyLastRound()) {
                correct.add(card);
            } else {
                incorrect.add(card);
            }
        }

        List<Card> result = new ArrayList<>();
        result.addAll(incorrect);
        result.addAll(correct);

        return result;
    }
}